doc/source/rllib/rllib-catalogs.rst

```diff
@@ -0,0 +1,151 @@
+    Besides these, we create a head network that fits these two by hand to show how you can combine RLLib's
+    ModelConfig API and Catalog. Extending Catalog to also build this head is how Catalog is meant to be
+    extended, which we cover later in this guide.
+
+    .. literalinclude:: ../../../rllib/examples/catalog/basics/cartpole_models.py
+       :language: python
+       :start-after: __sphinx_doc_begin__
+       :end-before: __sphinx_doc_end__
+
+What are Catalogs
+~~~~~~~~~~~~~~~~~
+
+Catalogs have two primary roles: Choosing the right model and choosing the right action distribution.
+By default, all catalogs implement decision trees that choose these based on a range of inputs.
+These include mainly the observation- and action-space of the RLModule, the model config dict and the tensor backend.
+Catalogs are also similar to what was called ModelCatalog in earlier versions of RLlib, but are more flexible.
+
+.. tabbed:: Catalogs vs RLModules
+
+    The following, broken down, diagram shows the general flow of information towards models and distributions.
+    RLModules create an instance of the Catalog class they receive as part of their constructor.
+    They then create their internal models and action distributions with the help of this Catalog.
+
+    .. note::
+      You can also modify the models and distributions in RLModules directly by overriding their constructor!
+
+    .. image:: images/catalog/catalog_and_rlm_diagram.svg
+        :align: center
+
+.. tabbed:: PPOCatalog vs PPORLModule
+
+    The following diagram shows a concrete case in more detail.
+    Therein, the PPOCatalog is fed an observation space, action space, a model config dict and the view requirements
+    of the RLModule. The model config dicts and the view requirements are only of interest in special cases, such as
+    recurrent networks or attention networks. The PPORLModule has four components that are created by the PPOCatalog:
+    Encoder, value function head, policy head and action distribution. You can find out more about this
+    distinction between these components in our section on Models.
+
+    .. image:: images/catalog/ppo_catalog_and_rlm_diagram.svg
+        :align: center
+
+Catalogs and AlgorithmConfig
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since Catalogs effectively control what models and distributions RLlib uses under the hood,
+they are also part of RLlib’s configurations. As the primary entry point for configuring RLlib,
+AlgorithmConfig is the place where you can configure the Catalogs of the RLModules that are created.
+The following example shows how to configure the Catalogs of the RLModules that are created by PPO.
+
+.. literalinclude:: ../../../rllib/examples/catalog/basics/catalogs_in_algo_configs.py
+    :language: python
+    :start-after: __sphinx_doc_begin__
+    :end-before: __sphinx_doc_end__
+
+.. image:: images/catalog/catalog_rlmspecs_diagram.svg
```

Comment:
It's for illustration, not for extra clarification.
When skimming docs, the majority of information I ingest is through pictures.
Basically, users can get an idea of what they are looking at just by looking at the pictures to filter information without putting in lots of effort through reading.
@angelinalg lmk if this is stupid, I'm open to deleting this if I have the wrong idea here.

Reply:
I think you need to add some extra caption to get the intention behind the the image across.

Reply:
Gotcha, I've updated the illustration a little bit and changed the text around it to get the point across that in case of heterogeneous agents, you'll want to modify the multi-agent rl module spec.

Reply:
Thanks!

Reply:
Generally it's good to have diagrams for visual learners.